gwas_sumstats_tools/gen_meta.py

```diff
@@ -0,0 +1,92 @@
+from pathlib import Path
+from rich import print
+from rich.progress import Progress, SpinnerColumn, TextColumn
+
+from gwas_sumstats_tools.interfaces.metadata import (
+    MetadataClient,
+    metadata_dict_from_gwas_cat,
+    get_file_metadata,
+)
+from gwas_sumstats_tools.utils import (
+    parse_accession_id,
+    append_to_path,
+    exit_if_no_data,
+)
+
+class Gen_meta:
+    def __init__(
+        self,
+        data_infile: Path,
+        metadata_infile: Path = None,
+        metadata_outfile: Path = None,
+        format_data: bool = False,
+    ) -> None:
+        self.format_data = format_data
+        self.data_infile = Path(data_infile)
+        self.metadata_outfile = Path(
+            self._set_metadata_outfile_name()
+            if not metadata_outfile
+            else metadata_outfile
+        )
+        self.meta = MetadataClient(
+            in_file=metadata_infile, out_file=self.metadata_outfile
+        )
+
+    def _set_metadata_outfile_name(self) -> str:
+        #return append_to_path(self.data_outfile, "-meta.yaml")
+        return append_to_path(self.data_infile, "-meta.yaml")
+
+    def set_metadata(
+        self, from_gwas_cat: bool = False, custom_metadata: dict = None
+    ) -> MetadataClient:
+        """Set metadata.
+        The hierarchy of where metadata is set is as follows:
+        1. custom_metadata map (overwrites anything below)
+        2. metadata from the GWAS Catalog
+        3. metadata from the original input file
+        4. metadata inferred from the datafile
+
+        Keyword Arguments:
+            from_gwas_cat -- update with data from GWAS catalog (default: {False})
+            custom_metadata -- Update with this map (default: {None})
+
+        Returns:
+            metadata object
+        """
+
+        meta_dict = get_file_metadata(
+            in_file=self.data_infile,
+            out_file=self.data_infile,
```

Comment:
In and Out files are the same. Is this intentional?

Reply:
get_file_metadata is a function defined in gwas_sumstats_tools/interfaces/metadata.py. This function is also used in the format.py, which needs input (unformatted file), output (formatted files) and meta_output (formatted files -meta.yaml). Since gen_meta is generating meta-yaml files for formatted files, input=output (formatted files). I know it is a bit confusing.